### ISSUE TITLE
The meaning of an "item" in E0121 should be elaborated on

### ISSUE THREAD

**Issue by skmendez** [rust-lang/rust#86005]
I tried this code:

```rust
fn main() {
    const BAD: [_; 3] = [1u8, 2u8, 3u8];
}
```

I expected to see this happen: an error which explains why this doesn't work with `const` even though it does work with `let`, like what would happen with the code:

```rust
fn main() {
    const BAD = 1u8;
}
```

Instead, this happened: E0121, which doesn't explain that the problem is that const's need to be explicitly typed, but instead refers to an "item signature", which is an unexplained term of art. The explanation in `rustc --explain E0121` does not elaborate on what an "item signature" means either.

### Meta

`rustc --version --verbose`:
Occurs on 1.52.1 stable and on 1.54.0 nightly.

**Comment by skmendez** [comment]
Now that I'm looking at it, a lot of errors use the term item, but place I can find a definition for what's an item is in in the [Rust Reference](https://doc.rust-lang.org/reference/items.html). I don't think that understanding terms used for the syntax of Rust should be required to comprehend an error message; as it stands I think it would be reasonable for a Rust newcomer to assume that an "item" could be any statement including variable assignment.
